docs/examples/vectorfields.py

```diff
@@ -0,0 +1,493 @@
+# $$
+# \mathbf{F}=\operatorname{grad} \Phi+\operatorname{rot} \Psi
+# $$
+# where
+# $$
+# \operatorname{grad} \Phi:=\left[\begin{array}{l}
+# \partial \Phi / \partial x^{(0)} \\
+# \partial \Phi / \partial x^{(1)}
+# \end{array}\right] \text { and } \operatorname{rot} \Psi:=\left[\begin{array}{c}
+# \partial \Psi / \partial x^{(1)} \\
+# -\partial \Psi / \partial x^{(0)}
+# \end{array}\right]
+# $$
+#
+# This is reminiscent of a 3 dimensional [Helmholtz decomposition](https://en.wikipedia.org/wiki/Helmholtz_decomposition).
+#
+# The 2 dimensional decomposition motivates a different approach: placing priors on $\Psi$ and $\Phi$, allowing us to make assumptions directly about fundamental properties of $\mathbf{F}$. If we choose independent GP priors such that $\Phi \sim \mathcal{G P}\left(0, k_{\Phi}\right)$ and $\Psi \sim \mathcal{G P}\left(0, k_{\Psi}\right)$, then $\mathbf{F} \sim \mathcal{G P} \left(0, k_\text{Helm}\right)$ (since acting linear operations on a GPs give GPs).
+#
+# For $\mathbf{X}, \mathbf{X}^{\prime} \in \mathbb{R}^2 \times \left\{0,1\right\}$ and $z, z^\prime \in \{0,1\}$,
+#
+# $$
+# \boxed{ k_{\mathrm{Helm}}\left(\mathbf{x}, \mathbf{x}^{\prime}\right)_{z,z^\prime} =  \frac{\partial^2 k_{\Phi}\left(\mathbf{x}, \mathbf{x}^{\prime}\right)}{\partial x^{(z)} \partial\left(x^{\prime}\right)^{(z^\prime)}}+(-1)^{z+z^\prime} \frac{\partial^2 k_{\Psi}\left(\mathbf{x}, \mathbf{x}^{\prime}\right)}{\partial x^{(1-z)} \partial\left(x^{\prime}\right)^{(1-z^\prime)}}}.
+# $$
+#
+# where $x^{(z)}$ and $(x^\prime)^{(z^\prime)}$ are the $z$ and $z^\prime$ components of $\mathbf{X}$ and ${\mathbf{X}}^{\prime}$ respectively.
+#
+# We compute the second derivatives using `jax.hessian`. In the following implementation, for a kernel $k(\mathbf{x}, \mathbf{x}^{\prime})$, this computes the Hessian matrix with respect to the components of $\mathbf{x}$
+#
+# $$
+# \frac{\partial^2 k\left(\mathbf{x}, \mathbf{x}^{\prime}\right)}{\partial x^{(z)} \partial x^{(z^\prime)}}.
+# $$
+#
+# Note that we have operated $\dfrac{\partial}{\partial x^{(z)}}$, *not* $\dfrac{\partial}{\partial \left(x^\prime \right)^{(z)}}$, as the boxed equation suggests. This is not an issue if we choose stationary kernels $k(\mathbf{x}, \mathbf{x}^{\prime}) = k(\mathbf{x} - \mathbf{x}^{\prime})$ , as the partial derivatives with respect to the components have the following exchange symmetry:
+#
+# $$
+# \frac{\partial}{\partial x^{(z)}} = - \frac{\partial}{\partial \left( x^\prime \right)^{(z)}}
+# $$
+#
+# for either $z$. We specify that this implementation only works with stationary kernels by setting the type of `kernelPhi` and `kernelPsi` to `gpx.kernels.stationary`.
+# %%
+
+
+@dataclass
+class HelmholtzKernel(gpx.kernels.AbstractKernel):
+    # initialise Phi and Psi kernels as any stationary kernel in gpJax
+    kernelPhi: gpx.kernels.stationary = gpx.kernels.RBF(active_dims=[0, 1])
```

Comment:
I think it would be better to have more informative names for these kernels! e.g. kernel_for_stream e.t.c